From c046c2a2c57587243b1fc53e65061f3e842848e3 Mon Sep 17 00:00:00 2001
From: Kunkun Jiang <jiangkunkun@huawei.com>
Date: Mon, 19 Jul 2021 11:21:02 +0800
Subject: [PATCH] iommu/arm-smmu-v3: Remove the redundant shift operation of
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c046c2a2c57587243b1fc53e65061f3e842848e3

 'size'

virt inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I401IF
CVE: NA

------------------------------

There is no reason to do the shift operation to 'size' in
arm_smmu_cache_invalidate().

Signed-off-by: Kunkun Jiang <jiangkunkun@huawei.com>
Reviewed-by: Keqian Zhu <zhukeqian1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
index 1aa879f4257e..ac3497180e0c 100644
--- a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
+++ b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
@@ -3621,7 +3621,6 @@ arm_smmu_cache_invalidate(struct iommu_domain *domain, struct device *dev,
 		    !(granule_size & smmu_domain->domain.pgsize_bitmap)) {
 			tg = __ffs(smmu_domain->domain.pgsize_bitmap);
 			granule_size = 1 << tg;
-			size = size >> tg;
 		}
 
 		arm_smmu_tlb_inv_range_domain(info->addr, size,
-- 
2.26.2

